server/src/http/background_refresh.rs

```diff
@@ -15,11 +15,13 @@ pub async fn poll_for_token_status(
             let mut write_lock = sink.write().await;
             match write_lock.validate(vec![token.clone()]).await {
                 Ok(validated_tokens) => {
-                    let sink_result = write_lock.sink_tokens(validated_tokens).await;
+                    let sink_result = write_lock.sink_tokens(validated_tokens.clone()).await;
                     if let Err(err) = sink_result {
                         warn!("Couldn't sink token result: {err:?}")
                     } else {
-                        let _ = feature_channel.send(token).await;
+                        for valid in validated_tokens {
+                            let _ = feature_channel.send(valid).await;
```

Comment:
Would it be better to only send in filtered "client" tokens to the channel instead of sending in everything and filtering on the `refresh_features` method?

Reply:
Occurred to me as well. I suspect we're going to tap into this from more than just this place, though, so it's nice to be defensive. Probably something for a future PR when the structure is more settled here if we only call this from the one place